### PR TITLE
Add subscription context

### PR DESCRIPTION
**Pull request by bvanneerven**
This PR tries to implement the functionality described in #151 .

### CODE REVIEW

File: lib/subscription-connection.js [hunk 1]
```diff
@@ -79,7 +81,8 @@ module.exports = class SubscriptionConnection {
 
     const sc = new SubscriptionContext({
       fastify: this.fastify,
-      pubsub: this.subscriber
+      pubsub: this.subscriber,
+      context: this.context
```
Comment (frikille) [inline lib/subscription-connection.js:85]: I think the context shouldn't be added here, but to the actual GraphQL context object (line 95). If we add it here the context object can be accessed in the subscriber function via the `pubsub` object (`context.pubsub.context`). If it's spread on the GraphQL context object it seems better I think (also in line with your original example in #151 )

Comment (bvanneerven) [reply lib/subscription-connection.js]: I agree! Implemented this in https://github.com/mcollina/fastify-gql/pull/153/commits/c7d4cc846181ed5711b412267926ec4bcb162091

**Comment by frikille** [comment]
This looks good to me!

Could you update the Readme (and the type definition) with this new option?

File: lib/subscription.js [hunk 2]
```diff
@@ -21,12 +21,18 @@ function createConnectionHandler ({ schema, subscriber, fastify, lruGatewayResol
       return
     }
 
+    let context = {}
+    if (subscriptionContextFn) {
+      context = await subscriptionContextFn(connection, request)
```
Comment (mcollina) [inline lib/subscription.js:26]: I would prefer if these would properties a single object instead.

Also, could you add a test where this function throws an error? Thanks

**Comment by katsanva** [comment]
@mcollina Hey. Is there something that prevents this PR to be merged?

**Comment by mcollina** [comment]
1. it's still flagged as "draft", so I thought @bvanneerven  were still working on it
2. this comment must be addressed https://github.com/mcollina/fastify-gql/pull/153#pullrequestreview-406968633.

**Comment by bvanneerven** [comment]
@mcollina Yes, I am still interested in pursuing this. I asked some questions a while ago based on your review, because I didn't fully understand it. Could you take a look at those?

**Comment by mcollina** [comment]
Can you link the comment? I don't seem to be able to find it.

**Comment by bvanneerven** [comment]
@mcollina I cannot really link the comment itself, but the file is here: https://github.com/mcollina/fastify-gql/pull/153/files/a712d11fdae7e81464e0635ff85ca8fc4eae489d#diff-49234199fc0c7c90d47d4a9524c21f0b

**Comment by mcollina** [comment]
I'm lost. What are the questions?

**Comment by bvanneerven** [comment]
@mcollina Sorry it has been a while, haven't been able to create time for this recently because of uni...
Anyways, here is a screenshot of the comment I see on my screen:

Screenshot_2020-07-08 Add subscription context by bvanneerven · Pull Request #153 · mcollina fastify-gql


Also, could you guide me in creating the failing test? I'm not so sure where and how is the best place to implement it.

**Comment by mcollina** [comment]
I'm sorry that tests are failing on master. I'll ask to rebase as soon as master is fixed, sorry for the incovenience.

**Comment by mcollina** [comment]
I have reverted the commit on master.

**Pull request closed without merging**